native/src/main/java/io/ballerina/stdlib/http/api/HttpService.java

```diff
@@ -73,6 +74,7 @@
     private String chunkingConfig;
     private String mediaTypeSubtypePrefix;
     private String introspectionResourcePath;
+    private boolean treatNilableAsOptional = true;
```

Comment:
Do we need treatNilableAsOptional in this class? Won't that be enough to cache in the resource?

Reply:
We get the configuration in `buildHttpService` method and set the value for resources from `updateResourceTree` method. If we remove this from service, we have to pass it through `buildHttpService`->`processResources`->`updateResourceTree`. WDYT?

Reply:
Resolved in [`a572748`](https://github.com/ballerina-platform/module-ballerina-http/pull/653/commits/a57274874a0c3a0bc6357de23c9053771bc96b6e)